pangeo_forge_recipes/transforms.py

```diff
@@ -598,6 +606,7 @@ class StoreToZarr(beam.PTransform, ZarrWriterMixin):
     dynamic_chunking_fn: Optional[Callable[[xr.Dataset], dict]] = None
     dynamic_chunking_fn_kwargs: Optional[dict] = field(default_factory=dict)
     attrs: Dict[str, str] = field(default_factory=dict)
+    consolidate_dimension_coordinates: bool = False
     consolidated_metadata: Optional[bool] = True
```

Comment:
```suggestion
```

See comment below re: boolean flags vs. composition. I feel like @sbquinlan's design suggestion is the right way to go here: modularity/composibility over branching/boolean flag logic.

Note if we go this route we want to also make the following change in the body of StoreToZarr below (but GitHub won't let me make a comment on these lines because the PR does not already change them):

```diff
target_store = schema | PrepareZarrTarget(
    target=self.get_full_target(),
    target_chunks=target_chunks,
    attrs=self.attrs,
-   consolidated_metadata=self.consolidated_metadata,
+   # In StoreToZarr, always default to unconsolidated. This leaves it up to the
+   # user whether or not they want to consolidate with ConsolidateMetadata(). Also,
+   # it prevents a broken/inconsistent state that could arise from metadata being
+   # consolidated here, and then falling out of sync with coordinates if
+   # ConsolidateDimensionCoordinates() is applied to the output of StoreToZarr().
+   consolidated_metadata=False,
    encoding=self.encoding,
)

```